JDK11 master ut failed

```
2018-09-19 03:25:53,034 WARN [org.apache.dubbo.remoting.exchange.support.DefaultFuture:received] -  [DUBBO] The timeout response finally returned at 2018-09-19 03:25:53.034, response Response [id=1039, version=null, status=20, event=false, error=null, result=RpcResult [result=Lower, exception=null]], channel: /172.17.0.6:49932 -> /172.17.0.6:9010, dubbo version: 2.7.0-SNAPSHOT, current host: 172.17.0.6
Tests run: 7, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 4.319 sec <<< FAILURE! - in org.apache.dubbo.rpc.protocol.dubbo.DubboProtocolTest
testDubboProtocolWithMina(org.apache.dubbo.rpc.protocol.dubbo.DubboProtocolTest)  Time elapsed: 1.315 sec  <<< ERROR!
org.apache.dubbo.rpc.RpcException: Invoke remote method timeout. method: enumlength, provider: dubbo://127.0.0.1:9010/org.apache.dubbo.rpc.protocol.dubbo.support.DemoService?client=mina, cause: Waiting server-side response timeout by scan timer. start time: 2018-09-19 03:25:51.798, end time: 2018-09-19 03:25:53.032, client elapsed: 0 ms, server elapsed: 1234 ms, timeout: 1000 ms, request: Request [id=1039, version=2.0.2, twoway=true, event=false, broken=false, data=RpcInvocation [methodName=enumlength, parameterTypes=[class [Lorg.apache.dubbo.rpc.protocol.dubbo.support.Type;], arguments=[[Lorg.apache.dubbo.rpc.protocol.dubbo.support.Type;@60e0cfe6], attachments={path=org.apache.dubbo.rpc.protocol.dubbo.support.DemoService, version=0.0.0}]], channel: /172.17.0.6:49932 -> /172.17.0.6:9010
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocolTest.testDubboProtocolWithMina(DubboProtocolTest.java:95)
Caused by: org.apache.dubbo.remoting.TimeoutException: Waiting server-side response timeout by scan timer. start time: 2018-09-19 03:25:51.798, end time: 2018-09-19 03:25:53.032, client elapsed: 0 ms, server elapsed: 1234 ms, timeout: 1000 ms, request: Request [id=1039, version=2.0.2, twoway=true, event=false, broken=false, data=RpcInvocation [methodName=enumlength, parameterTypes=[class [Lorg.apache.dubbo.rpc.protocol.dubbo.support.Type;], arguments=[[Lorg.apache.dubbo.rpc.protocol.dubbo.support.Type;@60e0cfe6], attachments={path=org.apache.dubbo.rpc.protocol.dubbo.support.DemoService, version=0.0.0}]], channel: /172.17.0.6:49932 -> /172.17.0.6:9010
	at org.apache.dubbo.rpc.protocol.dubbo.DubboProtocolTest.testDubboProtocolWithMina(DubboProtocolTest.java:95)







Recived callback: server invoke callback : arg:1537327557966
Tests run: 5, Failures: 0, Errors: 1, Skipped: 1, Time elapsed: 3.011 sec <<< FAILURE! - in org.apache.dubbo.rpc.protocol.dubbo.ExplicitCallbackTest
TestCallbackMultiInstans(org.apache.dubbo.rpc.protocol.dubbo.ExplicitCallbackTest)  Time elapsed: 1.737 sec  <<< ERROR!
java.lang.IllegalStateException: callback instance not found
```

https://api.travis-ci.org/v3/job/430358124/log.txt

--- a/dubbo-compatible/src/test/java/org/apache/dubbo/generic/GenericServiceTest.java
+++ b/dubbo-compatible/src/test/java/org/apache/dubbo/generic/GenericServiceTest.java
@@ -58,7 +58,7 @@ public class GenericServiceTest {
         DemoService server = new DemoServiceImpl();
         ProxyFactory proxyFactory = ExtensionLoader.getExtensionLoader(ProxyFactory.class).getAdaptiveExtension();
         Protocol protocol = ExtensionLoader.getExtensionLoader(Protocol.class).getAdaptiveExtension();
-        URL url = URL.valueOf("dubbo://127.0.0.1:5342/" + DemoService.class.getName() + "?version=1.0.0&generic=true");
+        URL url = URL.valueOf("dubbo://127.0.0.1:5342/" + DemoService.class.getName() + "?version=1.0.0&generic=true$timeout=3000");
         Exporter<DemoService> exporter = protocol.export(proxyFactory.getInvoker(server, DemoService.class, url));
         Invoker<GenericService> invoker = protocol.refer(GenericService.class, url);
 

--- a/dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/DubboProtocolTest.java
+++ b/dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/DubboProtocolTest.java
@@ -52,7 +52,7 @@ public class DubboProtocolTest {
     public void testDemoProtocol() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9020/" + DemoService.class.getName() + "?codec=exchange")));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9020/" + DemoService.class.getName() + "?codec=exchange")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9020/" + DemoService.class.getName() + "?codec=exchange").addParameter("timeout", 3000l)));
         assertEquals(service.getSize(new String[]{"", "", ""}), 3);
     }
 
@@ -60,7 +60,7 @@ public class DubboProtocolTest {
     public void testDubboProtocol() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName())));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName())));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName()).addParameter("timeout", 3000l)));
         assertEquals(service.enumlength(new Type[]{}), Type.Lower);
         assertEquals(service.getSize(null), -1);
         assertEquals(service.getSize(new String[]{"", "", ""}), 3);
@@ -71,7 +71,7 @@ public class DubboProtocolTest {
         assertEquals(set.iterator().next(), "aa");
         service.invoke("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "", "invoke");
 
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=netty")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=netty").addParameter("timeout", 3000l)));
         // test netty client
         StringBuffer buf = new StringBuffer();
         for (int i = 0; i < 1024 * 32 + 32; i++)
@@ -79,7 +79,7 @@ public class DubboProtocolTest {
         System.out.println(service.stringLength(buf.toString()));
 
         // cast to EchoService
-        EchoService echo = proxy.getProxy(protocol.refer(EchoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=netty")));
+        EchoService echo = proxy.getProxy(protocol.refer(EchoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=netty").addParameter("timeout", 3000l)));
         assertEquals(echo.$echo(buf.toString()), buf.toString());
         assertEquals(echo.$echo("test"), "test");
         assertEquals(echo.$echo("abcdefg"), "abcdefg");
@@ -90,7 +90,7 @@ public class DubboProtocolTest {
     public void testDubboProtocolWithMina() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName()).addParameter(Constants.SERVER_KEY, "mina")));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName()).addParameter(Constants.CLIENT_KEY, "mina")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName()).addParameter(Constants.CLIENT_KEY, "mina").addParameter("timeout", 3000l)));
         for (int i = 0; i < 10; i++) {
             assertEquals(service.enumlength(new Type[]{}), Type.Lower);
             assertEquals(service.getSize(null), -1);
@@ -105,7 +105,7 @@ public class DubboProtocolTest {
             service.invoke("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "", "invoke");
         }
 
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=mina")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=mina").addParameter("timeout", 3000l)));
         // test netty client
         StringBuffer buf = new StringBuffer();
         for (int i = 0; i < 1024 * 32 + 32; i++)
@@ -113,7 +113,7 @@ public class DubboProtocolTest {
         System.out.println(service.stringLength(buf.toString()));
 
         // cast to EchoService
-        EchoService echo = proxy.getProxy(protocol.refer(EchoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=mina")));
+        EchoService echo = proxy.getProxy(protocol.refer(EchoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName() + "?client=mina").addParameter("timeout", 3000l)));
         for (int i = 0; i < 10; i++) {
             assertEquals(echo.$echo(buf.toString()), buf.toString());
             assertEquals(echo.$echo("test"), "test");
@@ -126,11 +126,11 @@ public class DubboProtocolTest {
     public void testDubboProtocolMultiService() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName())));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName())));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + DemoService.class.getName()).addParameter("timeout", 3000l)));
 
         RemoteService remote = new RemoteServiceImpl();
         protocol.export(proxy.getInvoker(remote, RemoteService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + RemoteService.class.getName())));
-        remote = proxy.getProxy(protocol.refer(RemoteService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + RemoteService.class.getName())));
+        remote = proxy.getProxy(protocol.refer(RemoteService.class, URL.valueOf("dubbo://127.0.0.1:9010/" + RemoteService.class.getName()).addParameter("timeout", 3000l)));
 
         service.sayHello("world");
 
@@ -149,7 +149,7 @@ public class DubboProtocolTest {
     public void testPerm() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange")));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange").addParameter("timeout", 3000l)));
         long start = System.currentTimeMillis();
         for (int i = 0; i < 1000; i++)
             service.getSize(new String[]{"", "", ""});
@@ -160,7 +160,7 @@ public class DubboProtocolTest {
     public void testNonSerializedParameter() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange")));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange").addParameter("timeout", 3000l)));
         try {
             service.nonSerializedParameter(new NonSerialized());
             Assert.fail();
@@ -173,7 +173,7 @@ public class DubboProtocolTest {
     public void testReturnNonSerialized() throws Exception {
         DemoService service = new DemoServiceImpl();
         protocol.export(proxy.getInvoker(service, DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange")));
-        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange")));
+        service = proxy.getProxy(protocol.refer(DemoService.class, URL.valueOf("dubbo://127.0.0.1:9050/" + DemoService.class.getName() + "?codec=exchange").addParameter("timeout", 3000l)));
         try {
             service.returnNonSerialized();
             Assert.fail();

--- a/dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/ExplicitCallbackTest.java
+++ b/dubbo-rpc/dubbo-rpc-dubbo/src/test/java/org/apache/dubbo/rpc/protocol/dubbo/ExplicitCallbackTest.java
@@ -122,7 +122,7 @@ public class ExplicitCallbackTest {
 
     @Test
     public void TestCallbackMultiInstans() throws Exception {
-        initOrResetUrl(2, 1000);
+        initOrResetUrl(2, 3000);
         initOrResetService();
         IDemoCallback callback = new IDemoCallback() {
             public String yyy(String msg) {
@@ -143,14 +143,13 @@ public class ExplicitCallbackTest {
             Thread.sleep(500);
             demoProxy.unxxx2(callback);
             Assert.assertEquals(0, demoProxy.getCallbackCount());
-            System.out.println("");
 
             demoProxy.xxx2(callback2);
             Assert.assertEquals(1, demoProxy.getCallbackCount());
             Thread.sleep(500);
             demoProxy.unxxx2(callback2);
             Assert.assertEquals(0, demoProxy.getCallbackCount());
-            System.out.println("");
+
             demoProxy.xxx2(callback);
             Thread.sleep(500);
             Assert.assertEquals(1, demoProxy.getCallbackCount());

